Add NVIDIA DCGM and DCGM-exporter (prometheus)

de11n opened the pull request:
###### Description of changes

Add NVIDIA [DCGM](https://developer.nvidia.com/dcgm) and dcgm-exporter to Nixpkgs.

In order for the build to work, I had to also add `plog` and `tclap_1_4` which is a long-lived branch of `tclap`. I also fixed some minor errors in a couple other packages.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Commit:
jsoncpp: fix enableStatic

The workaround is no longer needed for static builds.

Commit:
libevent: fix sslSupport = false

Commit:
tclap: add 1.4 variant

Commit:
dcgm: init at 3.1.8

Commit:
prometheus-dcgm-exporter: init at 3.1.8-3.1.5

de11n commented:
cc @jonringer @kiskae

Kiskae commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,113 @@
+, python3
+, tclap_1_4
+
+# Allow builds targeting non-NixOS distros to adjust where libcuda.so is found
```
Do you have a practical example of this and why using `addOpenGLRunpath` directly wouldn't be enough?

Because afaik other distros rely on ldconfig and the FHS to locate libraries, not the library rpath.

de11n replied:
I am deploying this package to both NixOS machines and RHEL machines with Nix installed. On the RHEL machines, it's important that I have control of the *order* in which libraries get loaded. I need libcuda to be loaded from `/lib[64]` but I don't want anything else loaded from there. So this option allows the Nix glibc etc. to take precedence over any FHS paths. I can easily override this option to make a build variant that works on RHEL.

Kiskae replied:
I'm honestly not sure if you can use the RPATH to prevent that behavior since it only affects the load order for that specific dynamic library. Once you load something outside of the nix store it can load anything it wants. The only way to prevent this is to load the required libraries yourself before you load the external library.

de11n replied:
For so files yes, but for bins the RPATH is reliable unless `LD_*` things are in the mix... but in that scenario it's the caller's fault. ;)

Kiskae replied:
That honestly sounds better since `addOpenGLRunpath` should be enough for NixOS and gives a single parameter to override for your specific use-case.

Kiskae reviewed (commented):
In addition to the comments I'd like to see a `nixpkgs-review` result to see if this would cause unrelated rebuilds due to `jsoncpp`/`libevent` package changes. It shouldn't because the changes are all gated by parameters but its best to check.

Kiskae commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,113 @@
+, git
+, jsoncpp
+, libevent
+, libyamlcpp
```
More specifically it is an alias for `yaml-cpp` and ofborg evaluates without enabling aliases.

Kiskae commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,113 @@
+  };
+
+  # Add our paths to the CUDA paths so FindCuda.cmake can find them.
+  EXTRA_CUDA_PATHS = lib.strings.concatMapStringsSep "\n" mkAppendCudaPaths cudaToolkits;
```
It does look like that would be difficult to untangle, I thought it might be possible to substitute at use but `FindCMake` hardcodes the 3 versions to load.

Depending on how these proxies are linked it might be possible to make them use `cudaPackages.libcublas` instead of the full toolkit and cut down on the closure size.

de11n replied:
I could not find a way to do this.

de11n commented:
Noting that plog was added before this got merged: https://github.com/NixOS/nixpkgs/pull/238806

Commit:
cudatoolkit: fix builds for 10.*

libnvrtc.so is not found in the same location in all versions of
cudatoolkit.

Commit:
cudatoolkit: fix build for 12.0.1

Many Qt dependencies were added only for 12.1 and above, but 12.0.1 also
needs them.

de11n commented:
@Kiskae 

> In addition to the comments I'd like to see a `nixpkgs-review` result to see if this would cause unrelated rebuilds due to `jsoncpp`/`libevent` package changes. It shouldn't because the changes are all gated by parameters but its best to check.

```
4 packages marked as broken and skipped:
python310Packages.caffeWithCuda python310Packages.theanoWithCuda python311Packages.theanoWithCuda truecrack-cuda

8 packages failed to build:
cudaPackages.tensorrt cudaPackages.tensorrt_8_5_1 cudaPackages.tensorrt_8_5_2 cudaPackages.tensorrt_8_6_0 katagoTensorRT python310Packages.tensorflowWithCuda python310Packages.tensorrt python311Packages.tensorrt

34 packages built:
caffeWithCuda colmapWithCuda cudaPackages.cuda-samples cudaPackages.cudatoolkit cudaPackages.cutensor dcgm forge gpu-burn gpu-screen-recorder gpu-screen-recorder-gtk gromacsCudaMpi gwe hip-nvidia katagoWithCuda librealsenseWithCuda nvtop nvtop-nvidia prometheus-dcgm-exporter python310Packages.cupy python310Packages.jaxlibWithCuda python310Packages.numbaWithCuda python310Packages.pycuda python310Packages.pynvml python310Packages.pyrealsense2WithCuda python310Packages.torchWithCuda python311Packages.caffeWithCuda python311Packages.cupy python311Packages.jaxlibWithCuda python311Packages.pycuda python311Packages.pynvml python311Packages.pyrealsense2WithCuda tclap_1_4 xgboostWithCuda xpraWithNvenc
```

All the tensorrt failures are due to the fact that...
```
To use the TensorRT derivation, you must join the NVIDIA Developer Program...
```

Kiskae commented on pkgs/development/libraries/libevent/default.nix:
```diff
@@ -20,6 +20,8 @@ stdenv.mkDerivation rec {
     })
   ];
 
+  configureFlags = lib.optional (!sslSupport) [ "--disable-openssl" ];
```
Since this is using `lib.optional` this results in `[ [ "--disable-openssl" ] ]`. Might as well remove the additional brackets.

de11n commented:
@Kiskae Thank you for the review. What's left to get this merged?

Kiskae commented:
@de11n I don't actually have merge permissions, but @NixOS/cuda-maintainers do and they are probably the ones most affected by any changes so they should take a look.

samuela commented on pkgs/development/libraries/tclap-1.4/default.nix:
```diff
@@ -0,0 +1,45 @@
+}:
+stdenv.mkDerivation {
+  pname = "tclap";
+  version = "1.4-3feeb7b";
```
```suggestion
  version = "1.4.0-rc1";
```

de11n replied:
The official `1.4.0-rc1` tag doesn't have unit tests and is missing pkgconfig files. That is why I went with this slightly more recent version. If I use 1.4.0-rc1 I will have to apply patches similar to what this commit is providing. See here: https://github.com/mirror/tclap/compare/1.4.0-rc1..1.4

I will add a comment to that effect.

samuela commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+{ lib
+, callPackage
+, gcc11Stdenv
+, fetchFromGitHub
+, addOpenGLRunpath
+, catch2
+, cmake
+, cudaPackages_10_2
+, cudaPackages_11_8
+, cudaPackages_12
+, fmt_9
+, git
+, jsoncpp
+, libevent
+, plog
+, python3
+, tclap_1_4
+, yaml-cpp
```
nit: alphabetize

de11n replied:
I believe it is already alphabetized? I was under the impression that `lib`, `callPackage` and a few others were special cased to always be at the beginning.

samuela replied:
ah I see... we don't have an established style guide for this unfortunately. I often see `lib` and `stdenv` special cased, but the details are not worth bike shedding here

samuela commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+
+  # libcuda.so must be found at runtime because it is supplied by the NVIDIA
+  # driver.
+  postFixup = ''
+    find "$out/bin" "$out/lib" -type f -executable -print0 | while IFS= read -r -d "" f; do
+      if isELF "$f" && [[ $(patchelf --print-needed "$f" || true) == *libcuda.so* ]]; then
+        addOpenGLRunpath "$f"
+      fi
+    done
+  '';
```
should be able to replace all of this with autoAddOpenGLRunpathHook. See eg https://github.com/NixOS/nixpkgs/blob/3297ece10cb7cdcc5c166f2fdaeb8fc5f84d3ec6/pkgs/applications/science/misc/colmap/default.nix#L39

de11n replied:
I realize it doesn't *currently* matter because of the way this hook is written, but that would require that I use a hook from a *particular* version cudaPackage, when the DCGM build system explicitly uses 3 of them. I did not want to be biased.

samuela replied:
ah yes this is an unfortunate consequence of how `autoAddOpenGLRunpathHook` is packaged. for better or worse `autoAddOpenGLRunpathHook` is the same in each `cudaPackages` attr set. I would recommend just picking the latest one and pulling from there

samuela replied:
depending on more than one `cudaPackages` in a single derivation is quite out of the ordinary :P

de11n replied:
Fair enough. I can switch to the hook and just add a comment that it doesn't *actually* pick a version for you. Yes, I was very surprised to find that DCGM needs all three and I did try to change their build system to avoid that. However, as noted in my comments to @Kiskae's review, it is *not* a trivial change to their build system to achieve that.

de11n replied:
The good news is that DCGM's *runtime* dependency is actually pretty small.

de11n replied:
autoAddOpenGLRunpathHook has a bug when it encounters statically linked things, apparently. I don't want to rebuild the world so I left this in here with a comment. I did switch it in `dcgm-exporter` though.

samuela replied:
Hmm that's unfortunate... what was the error exactly? could you please create an issue for it? we should really get on top of that

de11n replied:
I will open a separate PR to fix the hook and remove this hack.

samuela replied:
even better!

samuela commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+    echo "$EXTRA_CUDA_PATHS"$'\n'"$(cat cmake/FindCuda.cmake)" > cmake/FindCuda.cmake
+  '';
+
+  hardeningDisable = [ "all" ];
```
why is this necessary?

de11n replied:
I tried disabling only some hardenings with no success. I did not see a pattern in Nixpkgs of documenting the exact failures that exist with each hardening mode. That would be a significant amount of work and it would likely bitrot very quickly.

samuela replied:
ok, fair enough

samuela commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+}:
+let
+  # Flags copied from DCGM's libevent build script
+  libevent-nossl = libevent.override { sslSupport = false; };
+  libevent-nossl-static = libevent-nossl.overrideAttrs (super: {
+    CFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    CXXFLAGS = "-Wno-cast-function-type -Wno-implicit-fallthrough -fPIC";
+    configureFlags = super.configureFlags ++ [ "--disable-shared" "--with-pic" ];
+  });
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
```
why are these build flags necessary? if at all possible, it is preferable to just use dependencies as they are default packaged

de11n replied:
DCGM is *very* particular about its build. Its entire build system is actually deterministic, but it uses Docker to achieve that. This makes it extremely particular about the exact build flags used for every dependency. These dependencies are built with these specific flags and I ran into build failures in other configurations. While it might be theoretically possible to find a different configuration that succeeds, this one exactly matches what upstream does in its build system so it is the most likely to succeed.

samuela commented on pkgs/servers/monitoring/prometheus/dcgm-exporter/default.nix:
```diff
@@ -0,0 +1,67 @@
+{ lib
+, buildGoModule
+, addOpenGLRunpath
+, fetchFromGitHub
+, dcgm
+, linuxPackages
```
nit: alphabetize

samuela commented on pkgs/servers/monitoring/prometheus/dcgm-exporter/default.nix:
```diff
@@ -0,0 +1,67 @@
+
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "dcgm-exporter";
```
```suggestion
    repo = pname;
```

samuela commented on pkgs/servers/monitoring/prometheus/dcgm-exporter/default.nix:
```diff
@@ -0,0 +1,67 @@
+  src = fetchFromGitHub {
+    owner = "NVIDIA";
+    repo = "dcgm-exporter";
+    rev = version;
```
```suggestion
    rev = "refs/tags/${version}";
```
being more explicit is preferred when specifying revs

samuela reviewed (commented):
Reviewing from the @NixOS/cuda-maintainers perspective. Would be good to get a review from maintainers of the other packages

jsoncpp: @ttuegel @cpages 
libevent: no maintainer specified
tclap: no maintainer specified

de11n commented on pkgs/development/libraries/jsoncpp/default.nix:
```diff
@@ -40,17 +40,18 @@ stdenv.mkDerivation rec {
-    ++ lib.optional (!enableStatic) "-DBUILD_STATIC_LIBS=OFF";
+    ++ lib.optional ((stdenv.buildPlatform != stdenv.hostPlatform) || secureMemory) "-DJSONCPP_WITH_TESTS=OFF";
 
   # this is fixed and no longer necessary in 1.9.5 but there they use
   # memset_s without switching to a different c++ standard in the cmake files
   postInstall = lib.optionalString enableStatic ''
+    cp lib/libjsoncpp.a "$out/lib/libjsoncpp_static.a"
     (cd $out/lib && ln -sf libjsoncpp_static.a libjsoncpp.a)
   '';
```
I don't understand the comment so I didn't change it. Whatever it meant, the static build is incorrect without this change.

samuela replied:
yeah unfortunately I find the comment quite tricky to understand past "this is fixed and no longer necessary in 1.9.5" as well... perhaps author @tfc can shed some light here?

tfc replied:
Ok, i understand why the comment seems incomprehensible now, as i have a bit timely distance to it myself. :D 

Looking at the latest version of jsoncpp, they are no longer giving the static version of jsoncpp a `_static` filename suffix. This was a very bad idea anyway, so they stopped doing that. (It's only done for MSVC builds).

I think in the current version, you don't need this `cp`/`ln` dance any longer and static linking will just work out of the box.

So please just drop these lines.

de11n replied:
You are right. We can drop the postInstall entirely. Thank you!

samuela commented on pkgs/development/libraries/tclap-1.2/default.nix:
gotcha, thanks for explaining! yeah that's the directory structure generally used in nixpkgs

de11n replied:
Done.

de11n commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+
+  jsoncpp-static = jsoncpp.override { enableStatic = true; };
+
+  cudaToolkits = [
+    {
+      version = "10";
+      toolkit = cudaPackages_10_2.cudatoolkit;
+    }
+    {
+      version = "11";
+      toolkit = cudaPackages_11_8.cudatoolkit;
+    }
+    {
+      version = "12";
+      toolkit = cudaPackages_12.cudatoolkit;
+    }
+  ];
```
Can you make some specific recommendations of how I would avoid this? [Here is how DCGM gets cudatoolkit](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/scripts/0080_cuda.sh).

samuela replied:
https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/scripts/0080_cuda.sh#L24-L39

indicates what redist packages you're looking for. you can find the whole list of nix packages available with some tab completion-fu: `nix-build -A cudaPackages.<TAB>`

then it's just a matter of picking the ones you need. you can check https://github.com/NixOS/nixpkgs/pull/217410 for an example migration `cudatoolkit -> redist packages`

de11n replied:
Thank you. I will give that a shot.

de11n replied:
I was able to get this working for cudaPackages_11 and cudaPackages_12. Sadly, cudaPackages_10 does not have redist packages broken out at this level, it seems. I've made the dependencies granular for 11 and 12, and left 10 as `cudatoolkit`.

de11n replied:
I should point out that the result of my work is clearly much less clean than the previous solution because DCGM's CMake assumes all headers in the same tree structure, but if `cudatoolkit` is eventually being deprecated, then it might be worth the added complexity.

samuela replied:
Yes, and this additional complexity will be alleviated shortly with https://github.com/NixOS/nixpkgs/pull/233581

de11n replied:
Amazing. Little did I know I am not the first to resort to symlinkJoin 😂

de11n commented on pkgs/os-specific/linux/dcgm/default.nix:
```diff
@@ -0,0 +1,108 @@
+    list(APPEND Cuda${version}_INCLUDE_PATHS "${toolkit}/include")
+    list(APPEND Cuda${version}_LIB_PATHS "${toolkit}/lib" "${toolkit.lib}/lib")
+  '';
+in gcc11Stdenv.mkDerivation rec {
```
gcc11 is specifically [used by upstream](https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/build.sh#L22) and I couldn't get it to build with vanilla `gcc`. Like I said, DCGM is *very* particular about its build setup. I was rather surprised I could get it working in Nix at all... 😅

samuela replied:
ah ok let's add that in a comment then! specifying a custom gcc version is a bit unusual

de11n replied:
Done.

de11n commented on pkgs/development/compilers/cudatoolkit/common.nix:
```diff
@@ -178,7 +178,7 @@ backendStdenv.mkDerivation rec {
 
   preFixup = ''
-    patchelf $out/lib64/libnvrtc.so --add-needed libnvrtc-builtins.so
+    patchelf $(find "$out" -type l -name libnvrtc.so -execdir readlink -f {} \; | sort -u) --add-needed libnvrtc-builtins.so
```
I can do that, although we'd need to check for the highest version first.

samuela replied:
oh yeah good point. `versions.major version == "10"` is prob the way to go... at least I'm guessing that the location is consistent across major versions

de11n replied:
In case you like it better, I'm leaving this option here. However, I plan to do the if-based one.
```bash
# pick the file that exists; if none exist patchelf will fail with missing argument
patchelf $(ls "$out/lib/a" "$out/lib/b" 2>/dev/null) --add-needed libnvrtc.so
```

de11n replied:
Going with `lib.versionOlder version "11"` as that is used elsewhere in the file.

SomeoneSerge replied:
Also consider using

```
while IFS= read ...; do
  ...
done < <(find ...)
```

instead of `$(find ...)`. This avoids spawning a sub-shell and should handle whitespaces correctly (although we don't expect any)

de11n replied:
The original line you're commenting on is no longer in this PR.

de11n commented:
@samuela I've made many of the suggested changes. I'm not sure why this PR is now causing a rebuild of so many packages.

samuela reviewed (approved):
cuda-related changes LGTm

ConnorBaker commented:
CUDA changes look good to me as well -- thank you for the CUDA Toolkit fix.

de11n commented:
I have [one follow-up task](https://github.com/NixOS/nixpkgs/pull/235024#discussion_r1268701481) on this, but it shouldn't block merge. Is there anything else needed to get this merged?

samuela commented:
Let's do one last nixpkgs-review with the latest changes and then check with jsoncpp maintainers @ttuegel @cpages if they have any concerns?

de11n commented:
Result of `nixpkgs-review` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>112 packages failed to build:</summary>
  <ul>
    <li>ciscoPacketTracer8</li>
    <li>jellyfin-mpv-shim</li>
    <li>jellyfin-mpv-shim.dist</li>
    <li>linuxKernel.packages.linux_4_14_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_4_19_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4_hardened.sysdig</li>
    <li>notepadqq</li>
    <li>python310Packages.dalle-mini</li>
    <li>python310Packages.dalle-mini.dist</li>
    <li>python310Packages.distrax</li>
    <li>python310Packages.distrax.dist</li>
    <li>python310Packages.elegy</li>
    <li>python310Packages.elegy.dist</li>
    <li>python310Packages.flax</li>
    <li>python310Packages.flax.dist</li>
    <li>python310Packages.optuna</li>
    <li>python310Packages.optuna.dist</li>
    <li>python310Packages.pywebview</li>
    <li>python310Packages.pywebview.dist</li>
    <li>python310Packages.rlax</li>
    <li>python310Packages.rlax.dist</li>
    <li>python310Packages.scikit-tda</li>
    <li>python310Packages.scikit-tda.dist</li>
    <li>python310Packages.treex</li>
    <li>python310Packages.treex.dist</li>
    <li>python310Packages.vqgan-jax</li>
    <li>python310Packages.vqgan-jax.dist</li>
    <li>python311Packages.argos-translate-files</li>
    <li>python311Packages.argos-translate-files.dist</li>
    <li>python311Packages.argostranslate</li>
    <li>python311Packages.argostranslate.dist</li>
    <li>python311Packages.baselines</li>
    <li>python311Packages.baselines.dist</li>
    <li>python311Packages.cleanlab</li>
    <li>python311Packages.cleanlab.dist</li>
    <li>python311Packages.ctranslate2</li>
    <li>python311Packages.ctranslate2.dist</li>
    <li>python311Packages.dalle-mini</li>
    <li>python311Packages.dalle-mini.dist</li>
    <li>python311Packages.distrax</li>
    <li>python311Packages.distrax.dist</li>
    <li>python311Packages.dm-haiku</li>
    <li>python311Packages.dm-haiku.dist</li>
    <li>python311Packages.dm-haiku.testsout</li>
    <li>python311Packages.edward</li>
    <li>python311Packages.edward.dist</li>
    <li>python311Packages.elegy</li>
    <li>python311Packages.elegy.dist</li>
    <li>python311Packages.fastai</li>
    <li>python311Packages.fastai.dist</li>
    <li>python311Packages.faster-whisper</li>
    <li>python311Packages.faster-whisper.dist</li>
    <li>python311Packages.flax</li>
    <li>python311Packages.flax.dist</li>
    <li>python311Packages.gpt-2-simple</li>
    <li>python311Packages.gpt-2-simple.dist</li>
    <li>python311Packages.jmp</li>
    <li>python311Packages.jmp.dist</li>
    <li>python311Packages.k-diffusion</li>
    <li>python311Packages.k-diffusion.dist</li>
    <li>python311Packages.langchain</li>
    <li>python311Packages.langchain.dist</li>
    <li>python311Packages.libretranslate</li>
    <li>python311Packages.libretranslate.dist</li>
    <li>python311Packages.mhcflurry</li>
    <li>python311Packages.mhcflurry.dist</li>
    <li>python311Packages.n3fit</li>
    <li>python311Packages.n3fit.dist</li>
    <li>python311Packages.numpyro</li>
    <li>python311Packages.numpyro.dist</li>
    <li>python311Packages.optuna</li>
    <li>python311Packages.optuna.dist</li>
    <li>python311Packages.pot</li>
    <li>python311Packages.pot.dist</li>
    <li>python311Packages.rlax</li>
    <li>python311Packages.rlax.dist</li>
    <li>python311Packages.skrl</li>
    <li>python311Packages.skrl.dist</li>
    <li>python311Packages.spacy</li>
    <li>python311Packages.spacy-loggers</li>
    <li>python311Packages.spacy-loggers.dist</li>
    <li>python311Packages.spacy-lookups-data</li>
    <li>python311Packages.spacy-lookups-data.dist</li>
    <li>python311Packages.spacy-transformers</li>
    <li>python311Packages.spacy-transformers.dist</li>
    <li>python311Packages.spacy.dist</li>
    <li>python311Packages.tensorflow (python311Packages.tensorflow-build ,python311Packages.tensorflowWithoutCuda)</li>
    <li>python311Packages.tensorflow.dist (python311Packages.tensorflow-build.dist ,python311Packages.tensorflowWithoutCuda.dist)</li>
    <li>python311Packages.tensorflow-probability</li>
    <li>python311Packages.tensorflow-probability.dist</li>
    <li>python311Packages.tensorflowWithCuda</li>
    <li>python311Packages.tensorflowWithCuda.dist</li>
    <li>python311Packages.textacy</li>
    <li>python311Packages.textacy.dist</li>
    <li>python311Packages.textnets</li>
    <li>python311Packages.textnets.dist</li>
    <li>python311Packages.tf2onnx</li>
    <li>python311Packages.tf2onnx.dist</li>
    <li>python311Packages.tflearn</li>
    <li>python311Packages.tflearn.dist</li>
    <li>python311Packages.translatehtml</li>
    <li>python311Packages.translatehtml.dist</li>
    <li>python311Packages.treex</li>
    <li>python311Packages.treex.dist</li>
    <li>python311Packages.trfl</li>
    <li>python311Packages.trfl.dist</li>
    <li>python311Packages.vqgan-jax</li>
    <li>python311Packages.vqgan-jax.dist</li>
    <li>python311Packages.wandb</li>
    <li>python311Packages.wandb.dist</li>
    <li>springLobby</li>
    <li>zombietrackergps</li>
  </ul>
</details>
<details>
  <summary>486 packages built:</summary>
  <ul>
    <li>CuboCore.corepdf</li>
    <li>PageEdit</li>
    <li>amarok</li>
    <li>amarok.doc</li>
    <li>anki</li>
    <li>anki.dist</li>
    <li>anki.doc</li>
    <li>anki.man</li>
    <li>arianna</li>
    <li>bottles</li>
    <li>bottles-unwrapped</li>
    <li>breath-theme</li>
    <li>caffeWithCuda</li>
    <li>caffeWithCuda.bin</li>
    <li>calamares</li>
    <li>calamares-nixos</li>
    <li>calibre</li>
    <li>cantor</li>
    <li>ceph</li>
    <li>ceph-client</li>
    <li>ceph-csi</li>
    <li>ceph.dev</li>
    <li>ceph.doc</li>
    <li>libceph (ceph.lib ,libceph.dev ,libceph.doc ,libceph.lib ,libceph.man)</li>
    <li>ceph.man</li>
    <li>clipgrab</li>
    <li>csound-qt</li>
    <li>cudaPackages.nsight_compute</li>
    <li>cutemarked-ng</li>
    <li>cutter</li>
    <li>dcgm</li>
    <li>deepin.deepin-reader</li>
    <li>deepin.deepin-voice-note</li>
    <li>digikam</li>
    <li>domoticz</li>
    <li>drogon</li>
    <li>eagle</li>
    <li>faust2sc</li>
    <li>fcitx5-chinese-addons</li>
    <li>fcitx5-mozc</li>
    <li>fluffychat</li>
    <li>fluffychat.debug</li>
    <li>foxdot (python310Packages.foxdot)</li>
    <li>foxdot.dist (python310Packages.foxdot.dist)</li>
    <li>freecad</li>
    <li>frescobaldi</li>
    <li>frescobaldi.dist</li>
    <li>frigate</li>
    <li>gamescope</li>
    <li>gitqlient</li>
    <li>globalprotect-openconnect</li>
    <li>golden-cheetah</li>
    <li>gpsbabel-gui</li>
    <li>gpsbabel-gui.doc</li>
    <li>gpt2tc</li>
    <li>gpu-screen-recorder</li>
    <li>gpu-screen-recorder-gtk</li>
    <li>headlines</li>
    <li>hqplayer-desktop</li>
    <li>hqplayer-desktop.doc</li>
    <li>huggle</li>
    <li>jami</li>
    <li>jellyfin-media-player</li>
    <li>jsoncpp</li>
    <li>jsoncpp.dev</li>
    <li>kbibtex</li>
    <li>kchmviewer</li>
    <li>kdeltachat</li>
    <li>kiwix</li>
    <li>kmymoney</li>
    <li>kodiPackages.pvr-hdhomerun</li>
    <li>kodiPackages.visualization-shadertoy</li>
    <li>labplot</li>
    <li>latte-dock</li>
    <li>lgogdownloader</li>
    <li>libgestures</li>
    <li>libjson-rpc-cpp</li>
    <li>liblinphone</li>
    <li>libpulsar</li>
    <li>libretranslate (python310Packages.libretranslate)</li>
    <li>libretranslate.dist (python310Packages.libretranslate.dist)</li>
    <li>libsForQt5.akonadi-calendar</li>
    <li>libsForQt5.akonadi-calendar-tools</li>
    <li>libsForQt5.akonadi-calendar-tools.dev</li>
    <li>libsForQt5.akonadi-calendar.dev</li>
    <li>libsForQt5.akonadi-contacts</li>
    <li>libsForQt5.akonadi-contacts.dev</li>
    <li>libsForQt5.akonadi-import-wizard</li>
    <li>libsForQt5.akonadi-import-wizard.dev</li>
    <li>libsForQt5.akonadiconsole</li>
    <li>libsForQt5.akregator</li>
    <li>libsForQt5.akregator.dev</li>
    <li>libsForQt5.alkimia</li>
    <li>libsForQt5.angelfish</li>
    <li>libsForQt5.aura-browser</li>
    <li>libsForQt5.bismuth</li>
    <li>libsForQt5.calendarsupport</li>
    <li>libsForQt5.calendarsupport.dev</li>
    <li>libsForQt5.discover</li>
    <li>libsForQt5.eventviews</li>
    <li>libsForQt5.eventviews.dev</li>
    <li>libsForQt5.falkon</li>
    <li>libsForQt5.ghostwriter</li>
    <li>libsForQt5.incidenceeditor</li>
    <li>libsForQt5.incidenceeditor.dev</li>
    <li>libsForQt5.kaccounts-providers</li>
    <li>libsForQt5.kaddressbook</li>
    <li>libsForQt5.kalarm</li>
    <li>libsForQt5.kalendar</li>
    <li>libsForQt5.kde-cli-tools</li>
    <li>libsForQt5.kdepim-addons</li>
    <li>libsForQt5.kdepim-runtime</li>
    <li>libsForQt5.kdeplasma-addons</li>
    <li>libsForQt5.kdev-php</li>
    <li>libsForQt5.kdev-python</li>
    <li>libsForQt5.kdevelop</li>
    <li>libsForQt5.kdevelop-unwrapped</li>
    <li>libsForQt5.kgpg</li>
    <li>libsForQt5.khotkeys</li>
    <li>libsForQt5.khotkeys.bin</li>
    <li>libsForQt5.khotkeys.dev</li>
    <li>libsForQt5.kinfocenter</li>
    <li>libsForQt5.kio-gdrive</li>
    <li>libsForQt5.kmail</li>
    <li>libsForQt5.kmail-account-wizard</li>
    <li>libsForQt5.kmail.doc</li>
    <li>libsForQt5.kmailtransport</li>
    <li>libsForQt5.kmailtransport.dev</li>
    <li>libsForQt5.kmenuedit</li>
    <li>libsForQt5.knotes</li>
    <li>libsForQt5.konqueror</li>
    <li>libsForQt5.kontact</li>
    <li>libsForQt5.korganizer</li>
    <li>libsForQt5.krohnkite</li>
    <li>libsForQt5.ksystemstats</li>
    <li>libsForQt5.kwin-dynamic-workspaces</li>
    <li>libsForQt5.kwin-tiling</li>
    <li>libsForQt5.kzones</li>
    <li>libsForQt5.libgravatar</li>
    <li>libsForQt5.libgravatar.dev</li>
    <li>libsForQt5.libkdepim</li>
    <li>libsForQt5.libkgapi</li>
    <li>libsForQt5.libksieve</li>
    <li>libsForQt5.libksieve.dev</li>
    <li>libsForQt5.libksysguard</li>
    <li>libsForQt5.libksysguard.bin</li>
    <li>libsForQt5.libksysguard.dev</li>
    <li>libsForQt5.libopenshot</li>
    <li>libsForQt5.mailcommon</li>
    <li>libsForQt5.mailcommon.dev</li>
    <li>libsForQt5.mailimporter</li>
    <li>libsForQt5.marble</li>
    <li>libsForQt5.marble.dev</li>
    <li>libsForQt5.mauikit-calendar</li>
    <li>libsForQt5.mbox-importer</li>
    <li>libsForQt5.messagelib</li>
    <li>libsForQt5.messagelib.dev</li>
    <li>libsForQt5.parachute</li>
    <li>libsForQt5.pim-data-exporter</li>
    <li>libsForQt5.pim-sieve-editor</li>
    <li>libsForQt5.pimcommon</li>
    <li>libsForQt5.pimcommon.dev</li>
    <li>libsForQt5.plasma-bigscreen</li>
    <li>libsForQt5.plasma-browser-integration</li>
    <li>libsForQt5.plasma-desktop</li>
    <li>libsForQt5.plasma-disks</li>
    <li>libsForQt5.plasma-disks.bin</li>
    <li>libsForQt5.plasma-disks.dev</li>
    <li>libsForQt5.plasma-mobile</li>
    <li>libsForQt5.plasma-remotecontrollers</li>
    <li>libsForQt5.plasma-systemmonitor</li>
    <li>libsForQt5.plasma-vault</li>
    <li>libsForQt5.plasma-workspace</li>
    <li>libsForQt5.plasma-workspace.dev</li>
    <li>libsForQt5.powerdevil</li>
    <li>libsForQt5.qt5.qtwebengine</li>
    <li>libsForQt5.qt5.qtwebengine.bin</li>
    <li>libsForQt5.qt5.qtwebengine.dev</li>
    <li>libsForQt5.qt5.qtwebview</li>
    <li>libsForQt5.qt5.qtwebview.bin</li>
    <li>libsForQt5.qt5.qtwebview.dev</li>
    <li>libsForQt5.systemsettings</li>
    <li>libsForQt5.systemsettings.bin</li>
    <li>libsForQt5.systemsettings.dev</li>
    <li>libsForQt5.tokodon</li>
    <li>libsForQt5.xdg-desktop-portal-kde</li>
    <li>libsForQt5.zanshin</li>
    <li>libtensorflow (libtensorflow.python)</li>
    <li>lightly-boehs</li>
    <li>lightly-qt</li>
    <li>linphone</li>
    <li>linuxKernel.packages.linux_4_14.sysdig</li>
    <li>linuxKernel.packages.linux_4_19.sysdig</li>
    <li>linuxKernel.packages.linux_5_10.sysdig</li>
    <li>linuxKernel.packages.linux_5_10_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_15.sysdig</li>
    <li>linuxKernel.packages.linux_5_15_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_5_4.sysdig</li>
    <li>linuxKernel.packages.linux_6_1.sysdig</li>
    <li>linuxKernel.packages.linux_hardened.sysdig (linuxKernel.packages.linux_6_1_hardened.sysdig)</li>
    <li>linuxKernel.packages.linux_6_3.sysdig</li>
    <li>linuxKernel.packages.linux_6_3_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_6_4.sysdig</li>
    <li>linuxKernel.packages.linux_6_4_hardened.sysdig</li>
    <li>linuxKernel.packages.linux_latest_libre.sysdig</li>
    <li>linuxKernel.packages.linux_libre.sysdig</li>
    <li>linuxKernel.packages.linux_lqx.sysdig</li>
    <li>linuxKernel.packages.linux_testing_bcachefs.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod.sysdig</li>
    <li>linuxKernel.packages.linux_xanmod_latest.sysdig (linuxKernel.packages.linux_xanmod_stable.sysdig)</li>
    <li>linuxKernel.packages.linux_zen.sysdig</li>
    <li>luminanceHDR</li>
    <li>maui-shell</li>
    <li>mellowplayer</li>
    <li>merkaartor</li>
    <li>mindforger</li>
    <li>minetest</li>
    <li>minetestserver</li>
    <li>mnemosyne</li>
    <li>mnemosyne.dist</li>
    <li>napari (python310Packages.napari)</li>
    <li>napari.dist (python310Packages.napari.dist)</li>
    <li>natron</li>
    <li>nextcloud-client</li>
    <li>nextcloud-client.dev</li>
    <li>nmapsi4</li>
    <li>nomachine-client</li>
    <li>onionshare-gui</li>
    <li>onionshare-gui.dist</li>
    <li>openai-full</li>
    <li>openai-full.dist</li>
    <li>openboard</li>
    <li>openshot-qt</li>
    <li>openshot-qt.dist</li>
    <li>openvpn3</li>
    <li>openvr</li>
    <li>p4v</li>
    <li>patray</li>
    <li>patray.dist</li>
    <li>pentestgpt</li>
    <li>pentestgpt.dist</li>
    <li>pentobi</li>
    <li>pingus</li>
    <li>plex-media-player</li>
    <li>polybarFull</li>
    <li>polychromatic</li>
    <li>prometheus-dcgm-exporter</li>
    <li>psi</li>
    <li>python-qt</li>
    <li>python310Packages.aeppl</li>
    <li>python310Packages.aeppl.dist</li>
    <li>python310Packages.aesara</li>
    <li>python310Packages.aesara.dist</li>
    <li>python310Packages.argos-translate-files</li>
    <li>python310Packages.argos-translate-files.dist</li>
    <li>python310Packages.argostranslate</li>
    <li>python310Packages.argostranslate.dist</li>
    <li>python310Packages.arviz</li>
    <li>python310Packages.arviz.dist</li>
    <li>python310Packages.augmax</li>
    <li>python310Packages.augmax.dist</li>
    <li>python310Packages.bambi</li>
    <li>python310Packages.bambi.dist</li>
    <li>python310Packages.baselines</li>
    <li>python310Packages.baselines.dist</li>
    <li>python310Packages.blackjax</li>
    <li>python310Packages.blackjax.dist</li>
    <li>python310Packages.caffeWithCuda</li>
    <li>python310Packages.caffeWithCuda.bin</li>
    <li>python310Packages.chex</li>
    <li>python310Packages.chex.dist</li>
    <li>python310Packages.cleanlab</li>
    <li>python310Packages.cleanlab.dist</li>
    <li>python310Packages.ctranslate2</li>
    <li>python310Packages.ctranslate2.dist</li>
    <li>python310Packages.dm-haiku</li>
    <li>python310Packages.dm-haiku.dist</li>
    <li>python310Packages.dm-haiku.testsout</li>
    <li>python310Packages.dm-sonnet</li>
    <li>python310Packages.dm-sonnet.dist</li>
    <li>python310Packages.edward</li>
    <li>python310Packages.edward.dist</li>
    <li>python310Packages.fastai</li>
    <li>python310Packages.fastai.dist</li>
    <li>python310Packages.faster-whisper</li>
    <li>python310Packages.faster-whisper.dist</li>
    <li>python310Packages.gpt-2-simple</li>
    <li>python310Packages.gpt-2-simple.dist</li>
    <li>python310Packages.jax</li>
    <li>python310Packages.jax.dist</li>
    <li>python310Packages.jaxlib (python310Packages.jaxlib-build ,python310Packages.jaxlibWithoutCuda)</li>
    <li>python310Packages.jaxlib.dist (python310Packages.jaxlib-build.dist ,python310Packages.jaxlibWithoutCuda.dist)</li>
    <li>python310Packages.jaxlibWithCuda</li>
    <li>python310Packages.jaxlibWithCuda.dist</li>
    <li>python310Packages.jaxopt</li>
    <li>python310Packages.jaxopt.dist</li>
    <li>python310Packages.jmp</li>
    <li>python310Packages.jmp.dist</li>
    <li>python310Packages.k-diffusion</li>
    <li>python310Packages.k-diffusion.dist</li>
    <li>python310Packages.langchain</li>
    <li>python310Packages.langchain.dist</li>
    <li>python310Packages.magicgui</li>
    <li>python310Packages.magicgui.dist</li>
    <li>python310Packages.mhcflurry</li>
    <li>python310Packages.mhcflurry.dist</li>
    <li>python310Packages.n3fit</li>
    <li>python310Packages.n3fit.dist</li>
    <li>python310Packages.napari-npe2</li>
    <li>python310Packages.napari-npe2.dist</li>
    <li>python310Packages.numpyro</li>
    <li>python310Packages.numpyro.dist</li>
    <li>python310Packages.objax</li>
    <li>python310Packages.objax.dist</li>
    <li>python310Packages.optax</li>
    <li>python310Packages.optax.dist</li>
    <li>python310Packages.optax.testsout</li>
    <li>python310Packages.pot</li>
    <li>python310Packages.pot.dist</li>
    <li>python310Packages.pymc</li>
    <li>python310Packages.pymc.dist</li>
    <li>python310Packages.pyqt6-webengine</li>
    <li>python310Packages.pyqt6-webengine.dev</li>
    <li>python310Packages.pyqt6-webengine.dist</li>
    <li>python310Packages.pyqtwebengine</li>
    <li>python310Packages.pyqtwebengine.dev</li>
    <li>python310Packages.pyqtwebengine.dist</li>
    <li>python310Packages.pyside2</li>
    <li>python310Packages.pyside2-tools</li>
    <li>python310Packages.pyside6</li>
    <li>python310Packages.pytensor</li>
    <li>python310Packages.pytensor.dist</li>
    <li>python310Packages.skrl</li>
    <li>python310Packages.skrl.dist</li>
    <li>python310Packages.spacy</li>
    <li>python310Packages.spacy-loggers</li>
    <li>python310Packages.spacy-loggers.dist</li>
    <li>python310Packages.spacy-lookups-data</li>
    <li>python310Packages.spacy-lookups-data.dist</li>
    <li>python310Packages.spacy-transformers</li>
    <li>python310Packages.spacy-transformers.dist</li>
    <li>python310Packages.spacy.dist</li>
    <li>spyder (python310Packages.spyder)</li>
    <li>spyder.dist (python310Packages.spyder.dist)</li>
    <li>python310Packages.tensorflow (python310Packages.tensorflow-build ,python310Packages.tensorflowWithoutCuda)</li>
    <li>python310Packages.tensorflow-bin</li>
    <li>python310Packages.tensorflow-bin.dist</li>
    <li>python310Packages.tensorflow.dist (python310Packages.tensorflow-build.dist ,python310Packages.tensorflowWithoutCuda.dist)</li>
    <li>python310Packages.tensorflow-datasets</li>
    <li>python310Packages.tensorflow-datasets.dist</li>
    <li>python310Packages.tensorflow-probability</li>
    <li>python310Packages.tensorflow-probability.dist</li>
    <li>python310Packages.tensorflowWithCuda</li>
    <li>python310Packages.tensorflowWithCuda.dist</li>
    <li>python310Packages.textacy</li>
    <li>python310Packages.textacy.dist</li>
    <li>python310Packages.textnets</li>
    <li>python310Packages.textnets.dist</li>
    <li>python310Packages.tf2onnx</li>
    <li>python310Packages.tf2onnx.dist</li>
    <li>python310Packages.tflearn</li>
    <li>python310Packages.tflearn.dist</li>
    <li>python310Packages.translatehtml</li>
    <li>python310Packages.translatehtml.dist</li>
    <li>python310Packages.treeo</li>
    <li>python310Packages.treeo.dist</li>
    <li>python310Packages.trfl</li>
    <li>python310Packages.trfl.dist</li>
    <li>python310Packages.umap-learn</li>
    <li>python310Packages.umap-learn.dist</li>
    <li>python310Packages.wandb</li>
    <li>python310Packages.wandb.dist</li>
    <li>python311Packages.augmax</li>
    <li>python311Packages.augmax.dist</li>
    <li>python311Packages.blackjax</li>
    <li>python311Packages.blackjax.dist</li>
    <li>python311Packages.caffeWithCuda</li>
    <li>python311Packages.caffeWithCuda.bin</li>
    <li>python311Packages.chex</li>
    <li>python311Packages.chex.dist</li>
    <li>python311Packages.foxdot</li>
    <li>python311Packages.foxdot.dist</li>
    <li>python311Packages.jax</li>
    <li>python311Packages.jax.dist</li>
    <li>python311Packages.jaxlib (python311Packages.jaxlib-build ,python311Packages.jaxlibWithoutCuda)</li>
    <li>python311Packages.jaxlib.dist (python311Packages.jaxlib-build.dist ,python311Packages.jaxlibWithoutCuda.dist)</li>
    <li>python311Packages.jaxlibWithCuda</li>
    <li>python311Packages.jaxlibWithCuda.dist</li>
    <li>python311Packages.jaxopt</li>
    <li>python311Packages.jaxopt.dist</li>
    <li>python311Packages.objax</li>
    <li>python311Packages.objax.dist</li>
    <li>python311Packages.optax</li>
    <li>python311Packages.optax.dist</li>
    <li>python311Packages.optax.testsout</li>
    <li>python311Packages.pyqt6-webengine</li>
    <li>python311Packages.pyqt6-webengine.dev</li>
    <li>python311Packages.pyqt6-webengine.dist</li>
    <li>python311Packages.pyqtwebengine</li>
    <li>python311Packages.pyqtwebengine.dev</li>
    <li>python311Packages.pyqtwebengine.dist</li>
    <li>python311Packages.pyside6</li>
    <li>python311Packages.pywebview</li>
    <li>python311Packages.pywebview.dist</li>
    <li>python311Packages.spyder</li>
    <li>python311Packages.spyder.dist</li>
    <li>python311Packages.treeo</li>
    <li>python311Packages.treeo.dist</li>
    <li>qemu_full</li>
    <li>qemu_full.debug</li>
    <li>qemu_full.ga</li>
    <li>qmapshack</li>
    <li>qolibri</li>
    <li>qsyncthingtray</li>
    <li>qt6.full</li>
    <li>qt6.qtwebengine</li>
    <li>qt6.qtwebengine.dev</li>
    <li>qt6.qtwebview</li>
    <li>qt6.qtwebview.dev</li>
    <li>qtcreator-qt6</li>
    <li>qutebrowser</li>
    <li>qutebrowser-qt6</li>
    <li>qutebrowser-qt6.dist</li>
    <li>qutebrowser.dist</li>
    <li>radiotray-ng</li>
    <li>renderdoc</li>
    <li>retext</li>
    <li>retext.dist</li>
    <li>rssguard</li>
    <li>rstudio</li>
    <li>rstudioWrapper</li>
    <li>samba4Full</li>
    <li>samba4Full.dev</li>
    <li>samba4Full.man</li>
    <li>sdrangel</li>
    <li>seafile-client</li>
    <li>semantik</li>
    <li>sigil</li>
    <li>skrooge</li>
    <li>sl1-to-photon</li>
    <li>smb3-foundry</li>
    <li>sonic-pi</li>
    <li>stellarium</li>
    <li>streamdeck-ui</li>
    <li>streamdeck-ui.dist</li>
    <li>stremio</li>
    <li>subsurface</li>
    <li>supercollider</li>
    <li>supercollider-with-plugins</li>
    <li>supercollider-with-sc3-plugins</li>
    <li>supercolliderPlugins.sc3-plugins</li>
    <li>supercollider_scel</li>
    <li>syncplay</li>
    <li>syncthingtray</li>
    <li>sysdig</li>
    <li>tageditor</li>
    <li>tagger</li>
    <li>tclap_1_4</li>
    <li>teamspeak_client</li>
    <li>teamviewer</li>
    <li>teamviewer.dev</li>
    <li>tellico</li>
    <li>texmaker</li>
    <li>tts</li>
    <li>tts.dist</li>
    <li>tumpa</li>
    <li>tumpa.dist</li>
    <li>vivisect</li>
    <li>vivisect.dist</li>
    <li>vnote</li>
    <li>wacomtablet</li>
    <li>waybar</li>
    <li>webmacs</li>
    <li>webmacs.dist</li>
    <li>whisper-ctranslate2</li>
    <li>whisper-ctranslate2.dist</li>
    <li>wyoming-faster-whisper</li>
    <li>wyoming-faster-whisper.dist</li>
    <li>xf86_input_cmt</li>
    <li>xkb-switch-i3</li>
    <li>yacas</li>
    <li>yacas-gui</li>
    <li>yuzu-early-access</li>
    <li>yuzu-mainline</li>
    <li>zeal</li>
    <li>zeal-qt6</li>
  </ul>
</details>

samuela commented:
nixpkgs-review looks reasonable to me. I'll go ahead and merge tomorrow unless anyone objects?

de11n commented:
@samuela Pushed a change to one of my commit messages which had leftover cruft from a rebase. (The comparison shows no code differences.)
